core/src/main/java/org/sonarsource/sonarlint/core/WsHelperImpl.java

```diff
@@ -85,4 +92,32 @@ public String generateAuthenticationToken(ServerConfiguration serverConfig, Stri
     SonarLintWsClient client = createClient(serverConfig);
     return generateAuthenticationToken(new ServerVersionAndStatusChecker(client), client, name, force);
   }
+
+  @Override
+  public boolean sendTelemetryData(TelemetryClientConfig clientConfig, TelemetryData data) {
+    try {
+      TelemetryPayload payload = new TelemetryPayload(data.daysSinceInstallation(), data.daysOfUse(), data.product(), data.version(), data.connectedMode());
+      String json = payload.toJson();
+
+      HttpConnector httpConnector = buildTelemetryClient(clientConfig);
+      PostRequest post = new PostRequest("");
+      post.setMediaType("application/json");
+      httpConnector.post(post, json).failIfNotSuccessful();
+      return true;
+    } catch (Exception e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)